api/types/restrictions.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// NewNetworkRestrictions creates a new NetworkRestrictions with the given name.
+func NewNetworkRestrictions() *NetworkRestrictions{
+	return &NetworkRestrictions{
```

Comment:
That would be V4 resource (V is the version of the API, not the version of the resource). @russjones we should explain it somewhere, I see other folks being confused as well.

V in resources is a version of the API. We released roles V4, therefore all new resources will be created with V4 version. Not ideal, but exists historic reasons.

Reply:
No problem, will fix. That said, sorry, this makes little sense. If the whole API version is bumped, wouldn't you have to rename all existing resources to the new version suffix by that logic?

Reply:
Ideally - yes, we'd bump the version API for all resources. However this would have required renaming every time we updated a single resource, so by convention unchanged resources of V3 are also V4 if the latest API version is V4.

Confusing, I know, we should improve in the future or at least explain cc @russjones @Joerger

Reply:
Why not just version the resources, not the whole API? If you're versioning the whole API, I suppose it should be reflected in the name of the service (https://github.com/gravitational/teleport/blob/master/api/client/proto/authservice.proto#L889).

Reply:
Sorry to add to the confusion here, but from my understanding the Version on the resource corresponds to the Resource version, not the API version.

Additionally, the API version is going to [follow lock step with the Teleport version](https://github.com/gravitational/teleport/blob/master/rfd/0010-api.md#versioning), so having the resource version tied to API version won't make much sense going forward.

@klizhentas I think that this may have warped from its original conception over time, partially due to the API changes I've been working on. Changes had to be made to fix [version validation consistency issues](https://github.com/gravitational/teleport/issues/5811). The Version and Version suffix are now always equal, except during version to version migrations (RoleV3 -> RoleV4 is an ongoing migration). Also, some new resources we've added are `V1` or `V2`, such as `ClusterAuditConfigV2` and `ClusterNetworkingConfigV2`. 

I agree that we should come up with and document a clear guideline around resource versions, they were also unclear to new contributors such as myself. For now, `NetworkRestrictions` could be made either `V4` or `V1` without any drawbacks.